pgzero/screen.py

```diff
@@ -45,6 +45,20 @@ def filled_circle(self, pos, radius, color):
         pos = round_pos(pos)
         pygame.draw.circle(self._surf, make_color(color), pos, radius, 0)
 
+    def polygon(self, points, color):
+        """Draw a polygon."""
+        if not isinstance(points, list):
```

Comment:
You should not do this check. Any function that accepts a list should accept an arbitrary iterable. We've asserted rects in some other methods because there it's possible to be specific - there is no "rect protocol" in Python.

What we should check is that each point is a two-element sequence of numbers. The check for that could actually go in `round_pos()`. It would be better to catch ValueError and TypeError and re-raise with better messages than to pre-validate the types of the arguments, because that is hard to do correctly. For example a valid number should be an instance of `numbers.Number` but the actual correct test is whether the type implements `__round__()` to return an int.)

Reply:
> I’m wondering if the type check on the points parameter is too restrictive.

Oh, I see you called this out yourself. Ok, I agree. 😃

Reply:
I changed the check to require an iterable. I put the check separately so it does not catch TypeErrors from round_pos.

``` python
def polygon(self, points, color):
    """Draw a polygon."""
    try:
        iter(points)
    except TypeError:
        raise TypeError("screen.draw.filled_polygon() requires an iterable of points to draw")
    points = [round_pos(point) for point in points]
    pygame.draw.polygon(self._surf, make_color(color), points, 1)
```

Reply:
Here's a proposal for checks in `round_pos`:

``` python
def round_pos(pos):
    """Round a tuple position so it can be used for drawing."""
    try:
        x, y = pos
    except TypeError:
        raise TypeError("Specify point as tuple (x, y).")
    x = round(x)
    y = round(y)
    if not (isinstance(x, int) and isinstance(y, int):
        raise TypeError("Coordinates must be numbers.")
    return round(x), round(y)
```

Reply:
The `Specify point as tuple (x, y).` looks good, but make sure you `raise ... from None` because the chained exceptions are confusing. Maybe include `repr(pos)` to indicate what you actually got.

In the second case I would favour EAFP, eg.

```python
try:
    return round(x), round(y)
except TypeError:
    raise TypeError("Coordinate values must be numbers (not {!r})".format(pos)) from None
```